rootston/xdg_shell_v6.c

```diff
@@ -287,9 +313,7 @@ static void handle_destroy(struct wl_listener *listener, void *data) {
 	wl_list_remove(&roots_xdg_surface->request_resize.link);
 	wl_list_remove(&roots_xdg_surface->request_maximize.link);
 	wl_list_remove(&roots_xdg_surface->request_fullscreen.link);
-	wl_list_remove(&roots_xdg_surface->view->link);
-	view_finish(roots_xdg_surface->view);
-	free(roots_xdg_surface->view);
+	view_destroy(roots_xdg_surface->view);
```

Comment:
Would it fit better if we added a `destroy()` method to the view interface which is called by `view_destroy()`? That seems to be how all the other interfaces work.

Reply:
The problem is that if `view_destroy()` emits the `destroy` event, then that has to happen before all the extra cleanup to avoid people accidently using the freed things in their destroy handler.